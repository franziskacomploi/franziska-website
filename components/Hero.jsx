import Image from 'next/image'
import React from 'react'
import heroImage from '../public/assets/hero-image.jpeg'
import hero from '../data/hero.json'

const Hero = ({ showBanner }) => {
  return (
    <>
      <div className="w-full flex flex-col lg:flex-row items-center justify-center gap-6 my-16">
        <div>
          <div className="text-2xl font-light">{hero.jobTitle}</div>
          <div className="headline-xl">
            {hero.firstName}
            <br />
            {hero.lastName}
          </div>
        </div>
        <div className="w-80 h-80 rounded-full bg-darkMint shadow flex flex-row items-center justify-center">
          <Image
            width={300}
            height={300}
            src={heroImage}
            alt="franziska-profile-image"
            className="rounded-full"
          />
        </div>
        <div className="border-l-2">
          <div className="content-big font-light py-4 ml-4 w-72">
            {hero.description}
          </div>
        </div>
      </div>
      {showBanner && (
        <div className="content-big flex flex-col md:flex-row gap-4 items-center justify-evenly bg-beige text-center pt-16">
          <div>
            {hero.feature1}
            <br />
            <span className="font-bold">{hero.basedIn}</span>
          </div>
          <div>
            {hero.feature2}
            <br />
            <span className="font-bold">{hero.experience}</span>
          </div>
          <div>
            {hero.feature3}
            <br />
            <span className="font-bold">{hero.focusOn}</span>
          </div>
        </div>
      )}
    </>
  )
}

export default Hero
