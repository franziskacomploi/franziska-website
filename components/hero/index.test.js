import { mount } from '@vue/test-utils';
import { it, expect, describe } from 'vitest';
import Hero from './index.vue';

describe('Hero', () => {
  it('should mount', () => {
    const wrapper = mount(Hero, { shallow: true });
    expect(wrapper.exists()).toBeTruthy();
  });

  it('should render correctly', () => {
    const wrapper = mount(Hero, { shallow: true });
    expect(wrapper.element).toMatchSnapshot();
  });
});
