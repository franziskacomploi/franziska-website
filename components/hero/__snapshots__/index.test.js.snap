// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero > should render correctly 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="w-full flex flex-col lg:flex-row items-center justify-center gap-6 mb-4 md:my-16"
  >
    <div>
      <p
        class="text-2xl font-light"
      >
        Software Engineer
      </p>
      <p
        class="headline-xl"
      >
        Franziska 
        <br />
         Comploi
      </p>
    </div>
    <div
      class="w-80 h-80 rounded-full bg-charcoal shadow flex flex-row items-center justify-center"
    >
      <img
        alt="franziska profile image"
        class="w-[300px] h-[300px] rounded-full"
        loading="lazy"
        src="/assets/images/hero-image.jpeg"
      />
    </div>
    <div
      class="border-l-2"
    >
      <p
        class="content-big font-light py-4 ml-4 w-72"
      >
        I'm a Software Engineer and in love with coffee, knitting and good books.
      </p>
    </div>
  </div>
  <div
    class="content-big flex flex-col md:flex-row gap-5 md:gap-4 items-center justify-evenly bg-cultured text-center pt-8 md:pt-16 px-10 md:px-auto"
  >
    <p>
      Based in 
      <br />
      <span
        class="font-bold"
      >
        Potsdam, Germany & Berlin
      </span>
    </p>
    <p>
      Experience 
      <br />
      <span
        class="font-bold"
      >
        2+ years
      </span>
    </p>
    <p>
      Focus on 
      <br />
      <span
        class="font-bold"
      >
        Data & Engineering
      </span>
    </p>
  </div>
  
</div>
`;
